src/components/menu/MenuItem.js

```diff
@@ -30,34 +30,42 @@ class MenuItem extends PureComponent {
     const color = destructive ? 'ruby' : disabled ? 'neutral' : 'teal';
     const tint = disabled && destructive ? 'light' : disabled || destructive ? 'dark' : 'darkest';
 
+    const boxProps = pickBoxProps(others);
+    const restProps = omitBoxProps(others);
+
     return (
-      <Box
-        {...others}
-        alignItems="center"
-        className={classNames}
-        data-teamleader-ui="menu-item"
-        display="flex"
-        element="li"
-        onClick={this.handleClick}
-        paddingHorizontal={3}
-        paddingVertical={2}
-      >
-        {icon && (
-          <Icon color={color} tint={tint} marginRight={3}>
-            {icon}
-          </Icon>
-        )}
-        <Box flex={1}>
-          {label && (
-            <TextBody color={color} tint={tint}>
-              {label}
-            </TextBody>
-          )}
-          {caption && (
-            <TextBody color="neutral" tint="darkest">
-              {caption}
-            </TextBody>
+      <Box {...boxProps} data-teamleader-ui="menu-item" display="flex" element="li">
+        <Box
+          onClick={this.handleClick}
+          alignItems="center"
+          className={classNames}
+          disabled={disabled}
+          display="flex"
+          element={element}
+          flex="1 0 auto"
+          paddingHorizontal={3}
+          paddingVertical={2}
+          textAlign="left"
+          {...(element === 'a' && disabled && { tabIndex: '-1' })}
+          {...restProps}
+        >
+          {icon && (
+            <Icon color={color} tint={tint} marginRight={3}>
+              {icon}
+            </Icon>
           )}
+          <Box flex="1 0 auto">
```

Comment:
Actually we only want to render this Box when there's a label or a caption right? 
```suggestion
          {label || caption (<Box flex="1 0 auto">
```

Reply:
Isn't the label required?

Reply:
Doesn't seem required, it's indeed not really a logical use-case, but yeah 🤷‍♂